src/util/simplify_expr.cpp

```diff
@@ -297,6 +297,78 @@ bool simplify_exprt::simplify_function_application(exprt &expr)
 
     return false;
   }
+  else if(func_id == ID_cprover_string_compare_to_func)
+  {
+    const refined_string_exprt &s1 =
+      to_string_expr(function_app.arguments().at(0));
+
+    const auto s1_data_opt = try_get_string_data_array(s1, ns);
+
+    if(!s1_data_opt)
+    {
+      return true;
+    }
+
+    const array_exprt &s1_data = s1_data_opt->get();
+
+    const refined_string_exprt &s2 =
+      to_string_expr(function_app.arguments().at(1));
+
+    const auto s2_data_opt = try_get_string_data_array(s2, ns);
+
+    if(!s2_data_opt)
+    {
+      return true;
+    }
+
+    const array_exprt &s2_data = s2_data_opt->get();
+
+    int res;
+
+    if(s1_data.operands().size() == s2_data.operands().size() &&
+       s1_data.operands() == s2_data.operands())
+    {
+      expr = from_integer(0, expr.type());
+    }
+    else
+    {
+      const bool first_shorter =
+        s1_data.operands().size() < s2_data.operands().size();
+
+      const exprt::operandst &ops1 =
+        first_shorter ? s1_data.operands() : s2_data.operands();
+
+      const exprt::operandst &ops2 =
+        first_shorter ? s2_data.operands() : s1_data.operands();
+
+      auto it_pair = std::mismatch(ops1.begin(), ops1.end(), ops2.begin());
+
+      if(it_pair.first == ops1.end())
+      {
+        res = -1;
+      }
+      else
+      {
+        mp_integer c1 =
+          numeric_cast_v<mp_integer>(to_constant_expr(*it_pair.first));
+        mp_integer c2 =
+          numeric_cast_v<mp_integer>(to_constant_expr(*it_pair.second));
+
+        if(c1 < c2)
+        {
+          res = -1;
+        }
+        else
+        {
+          res = 1;
+        }
```

Comment:
Nit pick: `res = c1 < c2 ? -1 : 1;` would condense 8 lines into a single one.

Reply:
Function rewritten and simplified